service/src/main/java/org/sunbird/actor/organisation/OrganisationManagementActor.java

```diff
@@ -469,15 +477,169 @@ private void assignKey(Request request) {
   private void removeUnusedField(Request request) {
     request.getRequest().remove(JsonKey.ENC_KEYS);
     request.getRequest().remove(JsonKey.SIGN_KEYS);
+    request.getRequest().remove(JsonKey.EXHAUST_ENCRYPTION_KEY);
     request.getRequest().remove(JsonKey.USER_ID);
   }
 
   private void addKeysToRequestMap(Request request) {
     List<String> encKeys = (List<String>) request.get(JsonKey.ENC_KEYS);
     List<String> signKeys = (List<String>) request.get(JsonKey.SIGN_KEYS);
+    List<String> exhaustKeys = new ArrayList<String>();
+
     Map<String, List<String>> keys = new HashMap<>();
     keys.put(JsonKey.ENC_KEYS, encKeys);
     keys.put(JsonKey.SIGN_KEYS, signKeys);
+
+    String orgId = (String) request.get(JsonKey.ID);
+    Map<String, List<String>> fetchedKeys = getKeysInDB(request, orgId);
+
+    if (fetchedKeys.containsKey(JsonKey.EXHAUST_ENCRYPTION_KEY)) {
+      exhaustKeys = fetchedKeys.get(JsonKey.EXHAUST_ENCRYPTION_KEY);
+      keys.put(JsonKey.EXHAUST_ENCRYPTION_KEY, exhaustKeys);
+    }
     request.getRequest().put(JsonKey.KEYS, keys);
   }
+
+  private void addEncryptionKey(Request request) throws IOException {
+    RequestContext context = request.getRequestContext();
+    String processId = ProjectUtil.getUniqueIdFromTimestamp(1);
+    Map<String, Object> req = (Map<String, Object>) request.getRequest().get(JsonKey.DATA);
+    String orgId = (String) req.get(JsonKey.ORGANISATION_ID);
```

Comment:
Add validation for orgid in request and rootorgid of the requestor is same or not.